File: Modules/User/Models/SubstanceCategory.php
<?php

namespace Modules\User\Models;

use App\Support\Traits\Models\HasTranslation;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\HasMany;

class SubstanceCategory extends Model
{
    use HasTranslation;

    protected $fillable = [
        'name_ar',
        'name_en',
        'slug',
        'is_active',
        'sort_order',
    ];

    protected $casts = ['is_active' => 'boolean'];

    public function substances(): HasMany
    {
        return $this->hasMany(Substance::class);
    }

    protected function getTranslationColumns(): array
    {
        return ['name'];
    }
}

File: Solid.Api/Database/Entities/SubstanceCategory.cs
namespace Solid.Api.Database.Entities;

public sealed class SubstanceCategory
{
    public long Id { get; set; }

    public string NameAr { get; set; } = string.Empty;

    public string NameEn { get; set; } = string.Empty;

    public string Slug { get; set; } = string.Empty;

    public bool IsActive { get; set; } = true;

    public byte SortOrder { get; set; }

    public DateTime? CreatedAt { get; set; }

    public DateTime? UpdatedAt { get; set; }

    public ICollection<Substance> Substances { get; set; } = [];

    public ICollection<Group> Groups { get; set; } = [];

    public ICollection<Recommendation> Recommendations { get; set; } = [];
}
